lib/quadquizaminos_web/live/contests_live.ex

```diff
@@ -0,0 +1,132 @@
+defmodule QuadquizaminosWeb.ContestsLive do
+  use Phoenix.LiveView
+  import Phoenix.HTML.Form
+  import Phoenix.LiveView.Helpers
+
+  alias Quadquizaminos.Repo
+
+  alias Quadquizaminos.{Contests, Contest}
+  alias Quadquizaminos.Util
+  alias QuadquizaminosWeb.Router.Helpers, as: Routes
+
+  def mount(_params, session, socket) do
+    :timer.send_interval(1000, self(), :timer)
+    QuadquizaminosWeb.Endpoint.subscribe("timer")
+
+    {:ok,
+     socket |> assign(payloads: [], started_contests: [], contests: Contests.list_contests())}
+  end
+
+  def handle_event("save", %{"key" => "Enter", "value" => contest_name}, socket) do
+    {:noreply, socket |> _create_contest(contest_name)}
+  end
+
+  def handle_event("start", %{"contest" => name}, socket) do
+    socket =
+      if name in Contests.active_contests_names() do
+        Contests.resume_contest(name)
+        socket
+      else
+        _start_contest(socket, name)
+      end
+
+    {:noreply, socket}
+  end
+
+  def handle_event("pause", %{"contest" => name}, socket) do
+    Contests.pause_contest(name)
+    {:noreply, socket}
+  end
+
+  def handle_event("reset", %{"contest" => name}, socket) do
+    Contests.reset_contest(name)
+    {:noreply, socket}
+  end
+
+  def handle_event("stop", %{"contest" => name}, socket) do
+    {:noreply, _end_contest(socket, name)}
+  end
+
+  def handle_info(:timer, socket) do
+    inactive_contest =
+      Enum.reject(socket.assigns.contests, fn contest ->
+        contest.name in Contests.active_contests_names()
+      end)
+
+    contests =
+      Contests.active_contests_names()
+      |> Enum.map(fn name ->
+        time = Contests.time_elapsed(name)
+        status = Contests.contest_status(name)
+
+        contest =
+          Enum.find(socket.assigns.contests, fn contest ->
+            contest.name == name
+          end)
+
+        %{contest | time_elapsed: time, status: to_string(status)}
+      end)
+
+    {:noreply, assign(socket, contests: contests ++ inactive_contest)}
+  end
+
+  defp contest(name, started_contests) do
+    Enum.find(started_contests, fn contest -> contest.name == name end)
+  end
+
+  defp _create_contest(socket, contest_name) do
+    contests = socket.assigns.contests
+
+    case Contests.create_contest(%{name: contest_name}) do
+      {:ok, contest} -> assign(socket, contests: contests ++ [contest])
+      _ -> socket
+    end
+  end
+
+  defp _start_contest(socket, name) do
+    contests = Enum.reject(socket.assigns.contests, fn contest -> contest.name == name end)
```

Comment:
What is the importance of removing the contest then putting it back into the socket again?

Reply:
We remove it first to update it  with time elapsed before putting it back.This is to avoid duplicate of the contest in the contest board .

Reply:
Is there any way you can update the entry rather than removing and re adding? 🤔

Reply:
Yes, it's possible. Check on the new changes made